src/CartesianProduct.jl

```diff
@@ -549,3 +550,36 @@ function vertices_list(cpa::CartesianProductArray{N}
 
     return vlist
 end
+
+
+function linear_map(M::AbstractMatrix{N},
+    cp::CartesianProductArray{N};
+    cond_tol::Number=DEFAULT_COND_TOL,
+    use_inv::Bool=!issparse(M)) where {N<:Real}
+    if !isinvertible(M; cond_tol=cond_tol)
+        error("linear maps for Cartesian product arrays need to be invertible")
+    end
+
```

Comment:
It is reasonable to assert here that the linear map is commensurate with the cartesian product: `size(M, 2) = dim(cp)`.